### PR TITLE
Debugging setup of VS Code section update

### PR DESCRIPTION
**Pull request by mustious**
Signed-off-by: Mustapha Abdullahi <mustious@gmail.com>

## Description

The earlier documentation will give some users difficulty in setting up VS Code debugging with Kedro as there has been changes in VS Code UI. This PR provides a more descriptive guide to the users in setting up debugging. Following this commit (focused on the debugging), more updates will be made on the overall Kedro with VS Code setup documentation to reflect changes of VS Code UI.

## Development notes

Only documentation markdown changed.

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Comment by astrojuanlu** [comment]
RTD failure seems unrelated: https://beta.readthedocs.org/projects/kedro/builds/22239120/#206508356--366

**Review by astrojuanlu — COMMENTED** [review]
Thanks for your PR @mustious! I left a comment

File: docs/source/development/set_up_vscode.md [hunk 1]
```diff
@@ -109,7 +109,7 @@ PYTHONPATH=C:/path/to/project/src;%PYTHONPATH%
 
 You can find [more information about setting up environmental variables in the VSCode documentation](https://code.visualstudio.com/docs/python/environments#_environment-variable-definitions-file).
 
-Go to **Debug > Add Configurations**.
+Click on the **Run and Debug** icon on the left activity bar(press `Cmd + Shift + D` for macOS). If there is no existing configuration click on **create a launch.json file** else click **Add Configuration**.
```
[the remark this answers is not in the document]
Comment (mustious) [reply docs/source/development/set_up_vscode.md]: Thanks for the kind response. I have made more changes. Kindly review :)

File: docs/source/development/set_up_vscode.md [hunk 2]
```diff
@@ -109,7 +109,8 @@ PYTHONPATH=C:/path/to/project/src;%PYTHONPATH%
 
 You can find [more information about setting up environmental variables in the VSCode documentation](https://code.visualstudio.com/docs/python/environments#_environment-variable-definitions-file).
 
-Go to **Debug > Add Configurations**.
+Click on the **Run and Debug** icon on the left activity bar(press `Cmd + Shift + D` for macOS). If there is no existing configuration, click on **create a launch.json file** else click on the Dropdown arrow in the top-left (shown below) and click **Add Configuration** .
```
Comment (astrojuanlu) [inline docs/source/development/set_up_vscode.md:112]:
```suggestion
Click on the **Run and Debug** icon on the left activity bar (press `Cmd + Shift + D` for macOS). If there is no existing configuration, click on **create a launch.json file** else click on the Dropdown arrow in the top-left (shown below) and click **Add Configuration** .
```

**Comment by astrojuanlu** [comment]
The DCO bot is failing because commit authorship email does not match signoff email, @mustious do you mind having a look at that? You did right by signing off the commits with `-s`, but I think you have your personal email configured globally for git (you can check by doing `git config --get-all user.email`).

**Review by stichbury — CHANGES_REQUESTED** [review]
Requested changes noted in comment on the file

File: docs/source/development/set_up_vscode.md [hunk 3]
```diff
@@ -109,7 +109,8 @@ PYTHONPATH=C:/path/to/project/src;%PYTHONPATH%
 
 You can find [more information about setting up environmental variables in the VSCode documentation](https://code.visualstudio.com/docs/python/environments#_environment-variable-definitions-file).
 
-Go to **Debug > Add Configurations**.
+Click on the **Run and Debug** icon on the left activity bar (press `Cmd + Shift + D` for macOS). If there is no existing configuration, click on **create a launch.json file** else click on the Dropdown arrow in the top-left (shown below) and click **Add Configuration** .
```
[the remark this answers is not in the document]
Comment (mustious) [reply docs/source/development/set_up_vscode.md:112]: > Is the text **create a launch.json file** really all lower case, or is there some capitalisation?

Yes @stichbury it is all lower case. Thank you :)
Screenshot 2023-10-19 at 15 45 46

[the remark this answers is not in the document]
Comment (mustious) [reply docs/source/development/set_up_vscode.md]: I just made an update changing "Dropdown" to "dropdown". Also, it is exactly "Add Configuration" in VS Code. Please see screenshot below. Perhaps add more images will be helpful in future documentation updates. Thanks very much for the comments @stichbury :)

Screenshot 2023-10-19 at 15 49 26

File: docs/source/development/set_up_vscode.md [hunk 4]
```diff
@@ -109,7 +109,8 @@ PYTHONPATH=C:/path/to/project/src;%PYTHONPATH%
 
 You can find [more information about setting up environmental variables in the VSCode documentation](https://code.visualstudio.com/docs/python/environments#_environment-variable-definitions-file).
 
-Go to **Debug > Add Configurations**.
+Click on the **Run and Debug** icon on the left activity bar (press `Cmd + Shift + D` for macOS). If there is no existing configuration, click on **create a launch.json file** else click on the dropdown arrow in the top-left (shown below) and click **Add Configuration** .
```
Comment (stichbury) [inline docs/source/development/set_up_vscode.md:112]:
```suggestion
Click on the **Run and Debug** icon on the left activity bar (press `Cmd + Shift + D` for macOS). If there is no existing configuration, click on **create a launch.json file** else click on the dropdown arrow in the top-left (shown below) and click **Add Configuration**.
```

**Review by stichbury — APPROVED** [review]
LGTM thanks @mustious ! 🌟

**Comment by astrojuanlu** [comment]
(I'm meeting with @mustious next week to see if we can understand what's going on with the signoff 😄)

**Comment by mustious** [comment]
> (I'm meeting with @mustious next week to see if we can understand what's going on with the signoff 😄)

Thanks so much @astrojuanlu for your assistance 🚀🎉

**Comment by astrojuanlu** [comment]
RTD problem is unrelated:

```
sphinx.errors.SphinxWarning: /home/docs/checkouts/readthedocs.org/user_builds/kedro/checkouts/3181/docs/source/nodes_and_pipelines/pipeline_registry.md:34:broken link: https://github.com/kedro-org/kedro-starters/blob/main/spaceflights/%7B%7B%20cookiecutter.repo_name%20%7D%7D/src/%7B%7B%20cookiecutter.python_package%20%7D%7D/pipeline_registry.py (404 Client Error: Not Found for url: https://github.com/kedro-org/kedro-starters/blob/main/spaceflights/%7B%7B%20cookiecutter.repo_name%20%7D%7D/src/%7B%7B%20cookiecutter.python_package%20%7D%7D/pipeline_registry.py)
```

We will fix it separately

**Pull request merged**